cpp/src/datasets/make_blobs.cu

```diff
@@ -14,35 +14,65 @@
  * limitations under the License.
  */
 
+#include "common/cumlHandle.hpp"
+
 #include "make_blobs.hpp"
+
 #include "random/make_blobs.h"
 
+#include "linalg/transpose.h"
+
 namespace ML {
 namespace Datasets {
 
 void make_blobs(const cumlHandle& handle, float* out, int* labels, int n_rows,
                 int n_cols, int n_clusters, const float* centers,
-                const float* cluster_std,
-                const float cluster_std_scalar, bool shuffle,
-                float center_box_min, float center_box_max,
-                uint64_t seed) {
-  MLCommon::Random::make_blobs(out, labels, n_rows, n_cols, n_clusters,
-                               handle.getDeviceAllocator(), handle.getStream(),
-                               centers, cluster_std, cluster_std_scalar,
-                               shuffle, center_box_min, center_box_max, seed);
+                const float* cluster_std, const float cluster_std_scalar,
+                bool shuffle, float center_box_min, float center_box_max,
+                uint64_t seed, bool col_major) {
+  if (!col_major) {
+    MLCommon::Random::make_blobs(
```

Comment:
Please do not duplicate this call in this function. Depending on whether transpose is needed or not, you can allocate a temporary buffer and pass it to `make_blobs`.

Reply:
The point of doing this here in the algorithms layer, rather than in the prims layer, is that transposing in Python for large arrays is not proving to be as trivial to do as it is for us to do in the CUDA layer. While I don't mind doing the transpose in the GTests, I made this change so that I could enable the cuda-based `make_blobs` Python to be used inside of our dask-based make_blobs implementation.

Reply:
Also, since i had the made the change to do the transpose in the algorithms layer, I just used the argument here, rather than manually transposing in the GTest for DBSCAN

Reply:
Can you expand on why the transpose is tricky in the python layer? I guess I have a concern that we'll end up having to implement lots of similar wrappers in the future (at least any data generators) that just transpose at the end, so maybe we could start thinking about ways to generalize.

Reply:
There are several primitives in cuML where we do this, actually. It's not because it's a data generator, or because it's exposed to Python, but rather because it's in row-major format and we need it to be in col-major in order to put the result in a Dask cuDF. 

This isn't the first time I've run into this issue w/ our `row_matrix` utility when loading extremely large arrays. IIRC, there's an issue w/ the threads per buffer that creates an upper bound to the number elements in the array.  It just seems trivial to do this in the C++ layer....

Reply:
Since I want to get this out the door for 0.10, and it seems rather controversial, I've removed the `make_blobs` change for now. I'd like to get our `cuml.dask.dataset.make_blobs` using our cuml version soon, though.